src/main/java/codechicken/nei/recipe/DefaultOverlayHandler.java

```diff
@@ -116,10 +116,11 @@ private void moveIngredients(GuiContainer gui, List<IngredientDistribution> assi
                 if (!slot.getHasStack() || !canMoveFrom(slot, gui)) continue;
 
                 ItemStack stack = slot.getStack();
+                int backupStackSize = stack.stackSize; // clickSlot can kill the reference
                 if (!canStack(stack, pstack)) continue;
 
                 FastTransferManager.clickSlot(gui, slot.slotNumber);
-                int amount = Math.min(transferCap - transferred, stack.stackSize);
+                int amount = Math.min(transferCap - transferred, backupStackSize);
```

Comment:
How about moving this call above `clickSlot`? That removes the need for `backupStackSize`

Reply:
I like that approach alot, does the same but saves the extra int. Thank you, I will commit it later today